src/sdl2/video.rs

```diff
@@ -1177,23 +1177,13 @@ impl Window {
         }
     }
 
-    pub fn surface<'a>(&'a self, _e: &'a EventPump) -> Result<&'a SurfaceRef, String> {
+    pub fn surface<'a>(&'a mut self, _e: &'a EventPump) -> Result<Surface<'a>, String> {
```

Comment:
Wow, good thing you read the SDL docs on that. This seems to be the only case where we really need the SurfaceRef. I am okay with undoing the removal of SurfaceRef in that case. I think combining Window+WindowRef is still valid since it was not necessary.